tests/acceptance/stepDefinitions/filesContext.js

```diff
@@ -1,10 +1,65 @@
 const { client } = require('nightwatch-api')
-const { When } = require('cucumber')
+const { Given, When, Then } = require('cucumber')
 
 When('the user browses to the files page',
   () => {
-    const filesPage = client
+    return client
       .page.filesPage()
-    return filesPage
       .navigate()
   })
+
+Then('the files table should be displayed',
+  () => {
+    const filesPage = client.page.filesPage()
+    return filesPage
+      .waitForElementVisible('@filesTable')
+  })
+
+Given('the user has browsed to the files page', function () {
+  return client
+    .page.filesPage()
+    .navigate()
+});
+
+When('the user creates a folder with the name {string} using the webUI', function (folderName) {
+  client
+    .page.filesPage()
+    .waitForElementVisible('@newFileMenuButton', 500000)
+    .click('@newFileMenuButton')
+    .waitForElementVisible('@newFolderButton')
+    .click('@newFolderButton')
+    .waitForElementVisible('@newFolderInput')
+    .setValue('@newFolderInput', folderName)
+    .click('@newFolderOkButton')
+
+  return client
+    .useXpath()
+    .waitForElementVisible('//div[contains(@class, "file-row-name")][text()="'+folderName+'"]')
```

Comment:
I think we can put this XPath in pageObject aswell.

Reply:
Also, this step doesn't work if the folder is not created for some reason.(e.g invalid folder name). So, we cannot use this `When` step in such scenarios as it is the job of `Then` step to verify the folder is created.